packages/core/src/messages/responses/item-type-api.ts

```diff
@@ -0,0 +1,78 @@
+import {SdkError} from '../../errors/errors';
+import {
+  ResponseBase,
+  ResponseError,
+  ResponseHit,
+  ResponseMiss,
+} from './response-base';
+import {_ItemType} from './grpc-response-types';
+
+/**
+ * Parent response type for a item type request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Hit}
+ * - {Miss}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof ItemType.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `ItemType.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+
+export abstract class Response extends ResponseBase {}
+
+class _Hit extends Response {
+  private readonly keyType: _ItemType;
+
+  constructor(keyType: _ItemType) {
+    super();
+    this.keyType = keyType;
+  }
+
+  /**
+   * Returns the type of key.
+   * @returns string
+   */
+  public getItemType(): _ItemType {
```

Comment:
Can you double check some functions like `ListLength` and see if we are using `get` prefix for the accessors?  I feel like we are usually just calling them `length()`, etc., in which case this would just be `itemType()` - it's important for us to try to keep these APIs consistent.